api/app/Repositories/Interfaces/IUserRepository.php

```diff
@@ -11,7 +12,29 @@ interface IUserRepository
      *
      * @param string $apiKey user api key
      * @return User|null user if found, otherwise null.
-     * @throws \Exception in case of DB connection failure.
+     * @throws Exception in case of DB connection failure.
      */
     public function getUserByApiKey(string $apiKey) : ?User;
+    
+    /**
+     * Get all users.
+     *
+     * @return User[]|null
+     * @throws Exception in case of DB connection failure.
+     */
+    public function getUsers() : ?array;
+    
+    /**
+     * @param int $userId
+     * @param User $user
+     * @return bool
+     */
+    public function updateUser(int $userId, User $user) : bool;
```

Comment:
Just update

Reply:
And why do you pass user id which is already contained in User model?

Reply:
fixed